kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/decode/MppdbDecodingPlugin.java

```diff
@@ -77,10 +75,18 @@ public AbstractWALEvent decode(final ByteBuffer data, final BaseLogSequenceNumbe
     }
     
     private AbstractWALEvent decodeDataWithTX(final String dataText) {
+        if (majorVersion < 3) {
+            return decodeVersionLessThan3(dataText);
+        } else {
+            return decodeVersionGreaterThan3(dataText);
+        }
```

Comment:
`decodeVersionLessThan3` could be `decodeSerially`, `decodeVersionGreaterThan3` could be `decodeParallelly`